### PR TITLE
Add postgres and implement routes

### PR DESCRIPTION
**Pull request by goodpilotfish**
Before approving, please check out this branch and test the database setup steps in the README 💪

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -6,12 +6,23 @@ generator client {
...
+    provider = "postgresql"
     url      = env("DATABASE_URL")
 }
 
-model Example {
+model Draw {
+    id          String   @id @default(cuid())
+    createdAt   DateTime @default(now())
+    updatedAt   DateTime @updatedAt
+    passphrase  String   @unique
```
Comment (goodpilotfish) [inline prisma/schema.prisma:17]: TODO: Will improve with an index in a future PR

Comment (goodpilotfish) [inline prisma/schema.prisma:17]: TODO: Hash passwords and create an index

**Review by goodpilotfish — COMMENTED** [review]
Additional comments

File: prisma/schema.prisma [hunk 2]
```diff
@@ -6,12 +6,24 @@ generator client {
...
+    // TODO: Hash PW and add index
+    passphrase  String   @unique
+    giver       String
+    receiver    String
+    group       Group @relation(fields: [groupId], references: [id])
+    groupId     String
+}
+
+model Group {
     id        String   @id @default(cuid())
```
[the remark this answers is not in the document]
Comment (hortenbach) [reply prisma/schema.prisma:26]: you sre unique

File: src/server/api/routers/group.ts [hunk 3]
```diff
@@ -1,35 +1,59 @@
...
+import type { Draw } from '~/server/lib/shuffle';
+
+interface DrawEnhanced {
+  giver: string;
+  receiver: string;
+  passphrase: string;
+}
 
 export const groupRouter = createTRPCRouter({
   postGroup: publicProcedure
```
Comment (hortenbach) [inline src/server/api/routers/group.ts:16]: Maybe you can define a return type here like:
https://stackoverflow.com/questions/13315131/enforcing-the-type-of-the-indexed-members-of-a-typescript-object

**Review by hortenbach — APPROVED** [review]
good job please see comments

**Pull request merged**